src/Microsoft.Win32.Registry/src/Microsoft/Win32/RegistryKey.cs

```diff
@@ -1123,14 +1123,7 @@ internal Object InternalGetValue(String name, Object defaultValue, bool doNotExp
                         if (datasize % 2 == 1)
                         {
                             // handle the case where the registry contains an odd-byte length (corrupt data?)
-                            try
-                            {
-                                datasize = checked(datasize + 1);
-                            }
-                            catch (OverflowException e)
-                            {
-                                throw new IOException(SR.Arg_RegGetOverflowBug, e);
-                            }
```

Comment:
I'm not understanding the logic around why these are safe to remove.  What guarantees that datasize < int.MaxValue?